app/services/recaptcha_annotator.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class RecaptchaAnnotator
+  attr_reader :assessment_id, :analytics
+
+  # See: https://cloud.google.com/recaptcha-enterprise/docs/reference/rest/v1/projects.assessments/annotate#reason
+  module AnnotationReasons
+    INITIATED_TWO_FACTOR = :INITIATED_TWO_FACTOR
+    PASSED_TWO_FACTOR = :PASSED_TWO_FACTOR
+  end
+
+  # See: https://cloud.google.com/recaptcha-enterprise/docs/reference/rest/v1/projects.assessments/annotate#annotation
+  module Annotations
+    LEGITIMATE = :LEGITIMATE
+    FRAUDULENT = :FRAUDULENT
+  end
+
+  def initialize(assessment_id:, analytics:)
+    @assessment_id = assessment_id
+    @analytics = analytics
+  end
+
+  def annotate(reason: nil, annotation: nil)
+    submit_annotation(reason:, annotation:) if FeatureManagement.recaptcha_enterprise?
+    log_analytics(reason:, annotation:)
+  end
+
+  private
+
+  BASE_ENDPOINT = 'https://recaptchaenterprise.googleapis.com/v1'
+
+  def submit_annotation(reason:, annotation:)
+    request_body = { annotation: }.compact
+    request_body[:reasons] = [reason] if reason
```

Comment:
since we already have the `.compact`, could make it one hash build
```suggestion
    request_body = {
      annotation:,
      reasons: (reason && [reason]),
    }.compact
```

Reply:
Updated in a2066585aa.